pkg/model/openstackmodel/servergroup.go

```diff
@@ -108,6 +105,10 @@ func (b *ServerGroupModelBuilder) buildInstances(c *fi.ModelBuilderContext, sg *
 			}
 			subnets = append(subnets, b.LinkToSubnet(s(fmt.Sprintf("%s.%s", subnet, b.ClusterName()))))
 		}
+		if len(ig.Spec.Zones) > 0 {
```

Comment:
what is difference to current mechanism? I do not see the difference except then we have subnets AND zones in instancegroup spec. But it works in similar way

Reply:
At least in our OpenStack subnets are not bound to availability zones, which means the  new mechanism could reduce the subnets to 1, because the zones are not extracted from them.

Reply:
And, the current implementation is forcing us to use the zone name as subnet name.